AutoRest/AutoRest.Core/ClientModel/Parameter.cs

```diff
@@ -28,6 +28,20 @@ public Parameter()
         public string Name { get; set; }
 
         /// <summary>
+        /// Gets or sets the parameter client (explicitly defined code generation) name.
+        /// </summary>
+        public string ClientName {
```

Comment:
With this design, each language will be able to opt-in the extensions if appropriate.